apps/passport-server/src/services/generic-issuance/genericIssuanceService.ts

```diff
@@ -241,6 +251,104 @@ export class GenericIssuanceService {
     return relevantCapability.checkin(req);
   }
 
+  public async getAllUserPipelineDefinitions(
+    userId: string
+  ): Promise<PipelineDefinition[]> {
+    // TODO: Add logic for isAdmin = true
+    return (await this.definitionDB.loadPipelineDefinitions()).filter((d) =>
+      this.userHasPipelineDefinitionAccess(userId, d)
+    );
+  }
+
+  private userHasPipelineDefinitionAccess(
+    userId: string,
+    pipeline: PipelineDefinition
+  ): boolean {
+    return (
+      pipeline.ownerUserId === userId || pipeline.editorUserIds.includes(userId)
+    );
+  }
+
+  public async getPipelineDefinition(
+    userId: string,
+    pipelineId: string
+  ): Promise<PipelineDefinition> {
+    const pipeline = await this.definitionDB.getDefinition(pipelineId);
+    if (!pipeline || !this.userHasPipelineDefinitionAccess(userId, pipeline))
+      throw new PCDHTTPError(404, "Pipeline not found or not accessible");
+    return pipeline;
+  }
+
+  public async upsertPipelineDefinition(
+    userId: string,
+    pipelineDefinition: PipelineDefinition
```

Comment:
for the next iteration, it'll be important to not let this become a security issue w/ users changing stuff like event ids arbitrarily

Reply:
https://github.com/proofcarryingdata/zupass/issues/1459